modules/scalacheck/src/weaver/scalacheck/CheckConfig.scala

```diff
@@ -1,26 +1,70 @@
 package weaver
 package scalacheck
 
-case class CheckConfig(
-    minimumSuccessful: Int,
-    maximumDiscardRatio: Int,
-    maximumGeneratorSize: Int,
-    perPropertyParallelism: Int,
-    initialSeed: Option[Long]
-) {
-  assert(maximumDiscardRatio >= 0)
-  assert(maximumDiscardRatio <= 100)
-  assert(minimumSuccessful > 0)
+import org.scalacheck.rng.Seed
+
+sealed abstract class CheckConfig private { outer =>
```

Comment:
Using a similar pattern than in scalacheck for `Test.Parameters`.
https://github.com/typelevel/scalacheck/blob/main/core/shared/src/main/scala/org/scalacheck/Test.scala#L23